### ISSUE TITLE
Create a workflow diagram of the user experience 

### ISSUE THREAD

**Issue by bonicim** [VOLTTRON/copier-poetry-volttron-agent#9]
The diagram will help clearly visualize the goal and purpose of the copier tool. It will also help orient development efforts to get this tool ready for users. 

UPDATE: Will add this diagram to the README to help provide background to people who want to contribute or maintain this repo.

**Comment by bonicim** [comment]
![Screen Shot 2022-03-21 at 11 15 50 AM](https://user-images.githubusercontent.com/6901848/159337972-8990a56c-d235-4d30-a84c-66310acd3500.png)

[Copier-workflow-draft.txt](https://github.com/VOLTTRON/copier-poetry-volttron-agent/files/8318263/Copier-workflow-draft.txt)

**Comment by bonicim** [comment]
@kefeimo Let me know what you think. It's a somewhat high-level view of what we tried doing last Friday.

**Comment by kefeimo** [comment]
![image](https://user-images.githubusercontent.com/28743873/159343262-0989037d-ed83-4786-bb5a-4448086a4c4d.png)
@bonicim The workflow diagram looks clear to me. Just some minor comments here:
- The agent-template-generation sounds like a process, instead of a role/participant.
- Can you also provide a quick reference of the symbol in the diagram? (Just for consistency sake, not for rigidity purpose.)

**Comment by bonicim** [comment]
Yes, agent-template-generation is a process; it's the Q&A that the user uses to create the agent template locally. 

The boxes don't represent a particular symbol. It's more like a lower-level description of what's happening between the user and the copier tool. I can update his workflow based on your feedback.

**Comment by bonicim** [comment]
Updated workflow. 

<img width="1283" alt="Screen Shot 2022-03-23 at 1 44 55 PM" src="https://user-images.githubusercontent.com/6901848/159792447-d478f7f2-4e9f-4e58-a143-baa3e4c09bde.png">

[Copier-workflow-draft (1) (1).txt](https://github.com/VOLTTRON/copier-poetry-volttron-agent/files/8336547/Copier-workflow-draft.1.1.txt)

**Comment by craigpnnl** [comment]
I don't understand....we shouldn't fork the repo we just use copier to create the agent directory...then push it into a git somewhere...for us it would be under VOLTTRON but for others it wouldn't be.

The user will not fork copier repo unless they are changing the template.  Copier will use that template and fill out the correct properties from the question answered to generate a new repository (NOT TIED to our copier repo).

Something like this
```
User->copier "Runs copier command"
copier -> promt "user answers questions"
copier -> files "copier generates files and folders"
User -> github "creates github repository"
User -> github "pushes new repo to created repository"
```

**Comment by bonicim** [comment]
> I don't understand....we shouldn't fork the repo we just use copier to create the agent directory...then push it into a git somewhere...for us it would be under VOLTTRON but for others it wouldn't be.
> 
> The user will not fork copier repo unless they are changing the template. Copier will use that template and fill out the correct properties from the question answered to generate a new repository (NOT TIED to our copier repo).
> 
> Something like this
> 
> ```
> User->copier "Runs copier command"
> copier -> promt "user answers questions"
> copier -> files "copier generates files and folders"
> User -> github "creates github repository"
> User -> github "pushes new repo to created repository"
> ```

The workflow that I depicted was based on my user experience when you guided me using copier the last time we all met. I was confused then about what was going; the main confusion was on how to use this tool from an external user perspective vs. a developer working on the copier tool itself. The workflow diagram reflects that confusion and is probably the reason why you don't understand. But based on your feedback, I now understand what the workflow should be for an external user as opposed to a developer working on copier. I will revise the diagram based on your feedback.

**Comment by bonicim** [comment]
![image](https://user-images.githubusercontent.com/6901848/160925467-c79806cd-2333-4503-ae90-d4146a2e7cc6.png)
[Copier-workflow-draftv2.txt](https://github.com/VOLTTRON/copier-poetry-volttron-agent/files/8383854/Copier-workflow-draftv2.txt)

**Comment by kefeimo** [comment]
Fixed by PR #17 
Manually closed.

**Comment by kefeimo** [comment]
# Recurring topics along with updating README. Reopen as needed
